### PR TITLE
system_variables: add tidb_union_concurrency

### PR DESCRIPTION
**Pull request by wshwsh12**
### What is changed, added or deleted? (Required)

Add system variable: tidb_union_concurrency

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by TomShawn** [comment]
@wshwsh12 Does the `dev` version need this variable?

**Comment by wshwsh12** [comment]
cc @XuHuaiyu

**Comment by wshwsh12** [comment]
> @wshwsh12 Does the `dev` version need this variable? Please also involve a technical review, thanks~

No. Only 4.0/3.0 need this variable.

**Review by XuHuaiyu — COMMENTED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -479,6 +479,24 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
...
+
+- 作用域：SESSION | GLOBAL
+- 默认值：4
+- 这个变量用于设置 window 算子的并行度。
+
+### `tidb_union_concurrency`
+
+- 作用域：SESSION | GLOBAL
+- 默认值：4
+- 这个变量用于设置 union 算子的并行度。
```
Comment (TomShawn) [inline system-variables.md:498]:
```suggestion
- 这个变量用于设置 `union` 算子的并发度。
```

File: system-variables.md [hunk 2]
```diff
@@ -479,6 +479,24 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
...
+
+- 作用域：SESSION | GLOBAL
+- 默认值：4
+- 这个变量用来设置 `Projection` 算子的并发度。
+
+### `tidb_window_concurrency` <span class="version-mark">从 v4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：4
+- 这个变量用于设置 window 算子的并行度。
```
Comment (TomShawn) [inline system-variables.md:492]:
```suggestion
- 这个变量用于设置 `window` 算子的并发度。
```

File: system-variables.md [hunk 3]
```diff
@@ -431,6 +410,27 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
 - 默认值：0
 - 这个变量用来设置是否在日志里记录所有的 SQL 语句。
 
+### `tidb_build_stats_concurrency`
+
+- 作用域：SESSION
+- 默认值：4
+- 这个变量用来设置 ANALYZE 语句执行时并发度。
```
Comment (TomShawn) [inline system-variables.md:417]:
```suggestion
- 这个变量用来设置 `ANALYZE` 语句执行时并发度。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**